lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1741,6 +1775,12 @@ def configure_disks(self, vm_obj):
                 diskspec = self.device_helper.create_scsi_disk(scsi_ctl, disk_index)
                 disk_modified = True
 
+            # increment index for next disk search
+            disk_index += 1
+            # index 7 is reserved to SCSI controller
+            if disk_index == 7:
+                disk_index += 1
+
```

Comment:
This should also be capped at 15 based on today's ESXi scsi controller rules, unless we want to just let VMware enforce it with a task error...

Reply:
I am planning to have that in this PR https://github.com/ansible/ansible/pull/33338/